neurox/data/writer.py

```diff
@@ -218,13 +219,13 @@ def close(self):
 
 
 class JSONActivationsWriter(ActivationsWriter):
-    def __init__(self, filename):
+    def __init__(self, filename, dtype='float32'):
         super().__init__(filename, filetype="json")
         if not self.filename.endswith(".json"):
             raise ValueError(
                 f"Output filename ({self.filename}) does not end with .json, but output file type is json."
             )
-
+        self.dtype = dtype
```

Comment:
This should be passed to the parent constructor (like `HDF5ActivationsWriter` `super()` call)

Reply:
I'll take care of this in a future commit